duckduckgo/duckduckgo-privacy-extension
# Don't upgrade IPs, localhost and local TLDs to HTTPS

**Reviewer:** @dharb 

## Description:
This adds a shortpath for IPs, `localhost` and local TLDs (as defined in https://tools.ietf.org/html/rfc6761) which are not upgradabe by SE.
This fixes a problem where user doesn't have access to WAN, wants to access their ruter and our extension keeps upgrading local IP url to HTTPS making it impossible to access it.

This PR also:
- upgrades us from tldjs to tldts (finally) - so that we detect IPv6
- fixes integration tests

## Steps to test this PR:
1. Go to http://1.1.1.1, http://localhost, http://[::1], http://dev.localhost, http://example.test
2. Check in the extension's console that: a) there is a message saying that those were not upgraded because host is an IP, local TLD or `localhost` b) there are no requests to SE service

## Automated tests:

###### Reviewer Checklist:


###### PR Author Checklist:
